module/applications/item/item-sheet.mjs

```diff
@@ -460,7 +460,11 @@ export default class ItemSheet5e extends ItemSheet {
       {
         name: "DND5E.AdvancementControlDuplicate",
         icon: "<i class='fas fa-copy fa-fw'></i>",
-        condition,
+        condition: li => {
+          const id = li[0].closest(".advancement-item")?.dataset.id;
+          const advancement = this.item.advancement.byId[id];
+          return condition(li) && advancement?.constructor.availableForItem(this.item);
```

Comment:
I added it to `createAdvancement` (since `duplicateAdvancement` just calls that anyway). If users want to do something not allowed, they can always bypass this method and do a straight `update`.